.github/workflows/build.yml

```diff
@@ -4,16 +4,23 @@ on:
   push:
     branches: main
 
+  pull_request:
+    branches: main
+
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [1.18]
+      fail-fast: true
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: setup go environment
+      - name: Setup Go ${{ matrix.go-version }}
         uses: actions/setup-go@v3
         with:
-          go-version: '1.18.1'
+          go-version: ${{ matrix.go-version }}
```

Comment:
Are you using the latest go version of `1.18`?

Reply:
I am using 1.18.1 locally but the build was based on 1.18.2. Is there a concern?

Reply:
Could you checkout the `check-latest` option for `actions/setup-go@v3`?

Reply:
Added